mobile/library/cc/engine_builder.h

```diff
@@ -76,6 +77,9 @@ class EngineBuilder {
   EngineBuilder& enablePlatformCertificatesValidation(bool platform_certificates_validation_on);
 
   EngineBuilder& enableDnsCache(bool dns_cache_on, int save_interval_seconds = 1);
+  // Set additional socket options on the upstream cluster outbound sockets.
+  EngineBuilder& setAdditionalSocketOptions(
+      const std::vector<envoy::config::core::v3::SocketOption>& socket_options);
```

Comment:
We could also consider making this pass-by-value and doing `std::move` in the impl, but I'm sure it doesn't matter :)

Reply:
I considered that, but the other APIs in this class that do similar assume we don't move the original passed-in list, so I thought to keep it like this for consistency.